pandas_flavor/register.py

```diff
@@ -1,5 +1,85 @@
 from functools import wraps
-from pandas.api.extensions import register_series_accessor, register_dataframe_accessor
+from pandas.api.extensions import (
+    register_series_accessor,
+    register_dataframe_accessor,
+)
+import inspect
+from contextlib import nullcontext
+
+method_call_ctx_factory = None
+
+
+def handle_pandas_extention_call(method, method_signature, obj, args, kwargs):
+    """
+    This function is called when the user calls the registered method on pandas dataframe object.
+    pandas extention mechanism passes args and kwargs of original method call as it was applied to obj
+
+    Implementation uses global var *method_call_ctx_factory"
+
+    a) case when *method_call_ctx_factory* is None
+    In this case the implementation calls the registered method with unmodified args and kwargs and returns underlying method result.
+
+    b) case when *method_call_ctx_factory* is not None
+    In this case *method_call_ctx_factory* expected to refer to the function to create the context object. The context object will be used
+    to process inputs and outputs of *method* call. It is also possible that the context object method *handle_start_method_call*
+    will modify original args and kwargs before *method* call.
```

Comment:
method_call_ctx_factory is to allow pluggable module to be attached to pandas_flavor. I guess we may reconsider usage of global var as module interface and introduce plugin api. It could be justified if you envision existence of pandas_flavor as independent module in long term.

The only purpose of method_call_ctx_factory introduction is to allow pandas_flavor registered method call args and returns to be processed in plugin package. I was thinking about writing chapter into **pandas_flavor** README or other doc describing details. 

Thinking this way pyjanitor users would not need to be aware of underlying pandas_flavor module extensibility achieved via method_call_ctx_factory introduction. The value of method_call_ctx_factory suppose to be set by other package initialization (like pyjviz). Log message about method_call_ctx_factory value being set seems to be sufficient even for advanced users. If pyjanitor user decides not to use pyjviz then method_call_ctx_factory will stay None and all method call details logic will be bypassed.